clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -122,15 +122,27 @@ class external_mem_img_desc {
 
   /// size setter
   /// \param [in] img_dims The dimensions of the imported image resource
-  void set_size(sycl::range<3> img_dims) { size = img_dims; }
+  void set_size(sycl::range<3> img_dims) {
+    for (int i = 0; i < 3; i++) {
+      if (img_dims[i] > 1) {
+        size[i] = img_dims[i];
```

Comment:
Why it can keep size[i]=0 if (img_dims[i]<=1)

Reply:
In SYCL, the default dim of resource is (0, 0, 0) where as in CUDA, it is (1, 1, 1)